app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,22 @@ def index
   def show
     @post = Post.includes(:author, comments: :author).find(params[:id])
   end
+
+  def new
+    @post = Post.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @user = current_user
+    @post.author = @user
+    if @post.save
+      flash[:success] = 'User posted successfully'
+      redirect_to user_posts_path(@user)
+    else
+      flash.now[:error] = 'Post not saved'
+      render :new, status: :unprocessable_entity
+    end
```

Comment:
### OPTIONAL

- [ ] It is always good practice to have strong parameters. This enhances the security of your application. Kindly learn more about it [here](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides`)